src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -0,0 +1,170 @@
+    if (!IsPresetHandlePresentInPresets(delegate, commandData.presetHandle))
+    {
+        commandObj->AddStatus(commandPath, Status::NotFound);
+        return true;
+    }
+
+    // If the thermostat suggestions list is full, return RESOURCE_EXHAUSTED.
+    if (delegate->GetNumberOfThermostatSuggestions() >= delegate->GetMaxThermostatSuggestions())
+    {
+        commandObj->AddStatus(commandPath, Status::ResourceExhausted);
+        return true;
+    }
+
+    // If the effective time in UTC is greater than current time in UTC plus 24 hours, return INVALID_COMMAND.
+    const uint32_t kSecondsInDay = 24 * 60 * 60;
+    if (!commandData.effectiveTime.IsNull() && (commandData.effectiveTime.Value() > currentTimestamp + kSecondsInDay))
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    uint8_t uniqueID = delegate->GetUniqueID();
+    ThermostatSuggestionStructWithOwnedMembers thermostatSuggestion;
+    thermostatSuggestion.SetUniqueID(uniqueID);
+    thermostatSuggestion.SetPresetHandle(commandData.presetHandle);
+
+    uint32_t effectiveTime = commandData.effectiveTime.ValueOr(currentTimestamp);
+    thermostatSuggestion.SetEffectiveTime(effectiveTime);
+
+    const uint32_t kSecondsInMinute = 60;
+    thermostatSuggestion.SetExpirationTime(effectiveTime + (commandData.expirationInMinutes * kSecondsInMinute));
+
+    CHIP_ERROR err = delegate->AppendToThermostatSuggestionsList(thermostatSuggestion);
+
+    if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+    {
+        ChipLogError(Zcl, "Failed to AppendToThermostatSuggestionsList with error: %" CHIP_ERROR_FORMAT, err.Format());
+    }
+
+    err = delegate->ReEvaluateCurrentSuggestion(currentTimestamp);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to ReEvaluateCurrentSuggestion with error: %" CHIP_ERROR_FORMAT, err.Format());
+    }
+
+    Commands::AddThermostatSuggestionResponse::Type response;
+    response.uniqueID = uniqueID;
+    commandObj->AddResponse(commandPath, response);
+    return true;
+}
+
+bool emberAfThermostatClusterRemoveThermostatSuggestionCallback(
+    CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+    const Commands::RemoveThermostatSuggestion::DecodableType & commandData)
+{
+    auto delegate = GetDelegate(commandPath.mEndpointId);
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        commandObj->AddStatus(commandPath, Status::InvalidInState);
+        return true;
+    }
+
+    CHIP_ERROR err = delegate->RemoveFromThermostatSuggestionsList(commandData.uniqueID);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to RemoveFromThermostatSuggestionsList with uniqueID: %d error: %" CHIP_ERROR_FORMAT,
+                     commandData.uniqueID, err.Format());
+        commandObj->AddStatus(commandPath, Status::NotFound);
```

Comment:
How do we know this is NOT_FOUND?  Could something else fail under there?

If RemoveFromThermostatSuggestionsList is expected to always succeed, except when there was nothing with that id, then it should probably be returning a boolean for "found this suggestion"...

Reply:
Matches the spec right now - 

. If there is a <<ref_ThermostatSuggestionStruct, entry>> in the <<ref_ThermostatSuggestions>> attribute, whose <<ref_ThermostatSuggestionStructUniqueID>> field matches the <<ref_RemoveThermostatSuggestionUniqueID>> field passed in this command,

.. The server SHALL remove the <<ref_ThermostatSuggestionStruct, entry>> from the corresponding list.
.. The server SHALL return a status code of SUCCESS.
. Otherwise, the server SHALL return a status code of NOT_FOUND.

Reply:
Added to https://github.com/project-chip/connectedhomeip/issues/39943